test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2244,6 +2244,39 @@ describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
         expect(error).to.have.property('code', 'LAMBDA_FILE_SYSTEM_CONFIG_MISSING_VPC');
       });
     });
+
+    it('should throw when invalid tags are configured', async () => {
+      // The regular expression used to guard against invalid tags has had
+      // some issues, this test is here to prevent regressions
+      const invalidTagKeys = [
+        'aws:foo:bar', // aws: prefix
+        'foo;bar', // Used to work due to a bug in the regular expression
+        'foo@bar', // Would normally work, but not in CloudFormation templates
+        'a'.repeat(129), // Too long
+      ];
+
+      for (const tagKey of invalidTagKeys) {
+        try {
+          await runServerless({
+            fixture: 'function',
+            configExt: {
+              provider: {
+                stackTags: {
+                  [tagKey]: 'foo',
+                },
+              },
+            },
+            cliArgs: ['package'],
+          });
+
+          expect.fail('an error should have been thrown');
```

Comment:
Let's not confirm on specific schema validation errors in tests (otherwise we'll end with 10k+ scenarios to confirm). We have a sanity check that validation engine is configured properly, and schema engine on its own is tested extensively. I believe that's good enough.

What we should test here is simply that configuring tag as `foo:bar-baz` works as expected, and as I indicated in other comment, we already test function tagging here: https://github.com/serverless/serverless/blob/420e93740f1e9bffc285559b2567379f550f28af/test/unit/lib/plugins/aws/package/compile/functions.test.js#L379-L414

Let's simply add there such tag, and confirm it got configured as expected in CF template

Reply:
Adding a new tag here accomplishes nothing - the test passes regardless of whether the regular expression is fixed or not.

Reply:
Sorry, I didn't look carefully, and pasted you link to wrong test (one to be discarded).

Just add tag here: https://github.com/serverless/serverless/blob/7d2ca04d7f72981c422fb8f9c25ae4699aedffba/test/unit/lib/plugins/aws/package/compile/functions.test.js#L1466-L1469

Fact that it works will be automatically confirmed here: https://github.com/serverless/serverless/blob/7d2ca04d7f72981c422fb8f9c25ae4699aedffba/test/unit/lib/plugins/aws/package/compile/functions.test.js#L1553-L1562